controllers/slicegateway/slicegateway.go

```diff
@@ -1534,3 +1547,54 @@ func (r *SliceGwReconciler) ReconcileIntermediateGatewayDeployments(ctx context.
 
 	return ctrl.Result{}, nil, false
 }
+
+// createPodDisruptionBudgetForSliceGatewayPods checks for PodDisruptionBudget objects in the cluster that match the
+// slice gateway pods, and if missing, it creates a PDB with minimum availability of 1 so at least one pod remains in
+// case of a disruption.
+func (r *SliceGwReconciler) createPodDisruptionBudgetForSliceGatewayPods(ctx context.Context, log logr.Logger,
+	client client.Client, sliceName string, sliceGateway *kubeslicev1beta1.SliceGateway) error {
+	log = log.WithValues("sliceName", sliceName, "sliceGwName", sliceGateway.Name)
+
+	// List PDBs in cluster that match the slice gateway pods
+	pdbs, err := listPodDisruptionBudgetForSliceGateway(ctx, client, sliceName, sliceGateway.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "failed to list PodDisruptionBudgets that match the slice gateway")
+
+		// When some unexpected error occurred, return the error for requeuing the request
+		return err
+	}
+
+	// Check if PDB already exists that matches the current slice gateway
+	if len(pdbs) > 0 {
+		log.Info("PodDisruptionBudget matching the slice gateway already exists. Skipping creation.")
+
+		return nil
+	}
+
+	// Create PDB manifest with minimum availability of 1 pod
+	pdb := formatPodDisruptionBudget(sliceName, sliceGateway.Name, intstr.FromInt(1))
+
+	// Set SliceGateway instance as the owner and controller for PDB
+	ctrl.SetControllerReference(sliceGateway, pdb, r.Scheme)
+
+	// Create PDB for slice gateway's pod to have at least 1 pod on each worker when disruption occurs
+	if err = client.Create(ctx, pdb); err != nil {
```

Comment:
You can use `r.Create` instead of `r.Client.Create`

Reply:
You are right. `createPodDisruptionBudgetForSliceGatewayPods` was a function in the initial plan. I changed it to a method in the latest iteration but forgot to use the client from the receiver. Done now. Thanks!